### DISCUSSION TITLE
Quick notes from another DTV fork ...

### DISCUSSION THREAD

**Discussion by jasongdove** [chrisbenincasa/tunarr#188] [Ideas]
... in case you find them useful.

I experimented with a DTV fork at https://github.com/jasongdove/dizquetv that would incorporate the streaming/normalization logic from ETV.

ETV's logic can be found here: https://github.com/ErsatzTV/ErsatzTV/tree/main/ErsatzTV.FFmpeg
My quick attempt at porting some of that to TS can be found here: https://github.com/jasongdove/ffmpeg-pipeline

Partially incorporated here https://github.com/jasongdove/dizquetv/blob/1b29509b61f2d5b43cc1306e429832b76862f510/src/ffmpeg.js#L582-L672

I also created a bunch of dockerfiles to provide equivalent artifacts/tags (to ETV) https://github.com/jasongdove/dizquetv/tree/main/docker that are built with github workflows here https://github.com/jasongdove/dizquetv/blob/main/.github/workflows/docker.yml - these also use base images from ETV's ffmpeg builds here https://github.com/ErsatzTV/ErsatzTV-ffmpeg

In the end, I ended up abandoning all of this because it's easier for ETV to build a channel from DTV's channel JSON (and therefore use ETV's normalization pipeline natively) than it is to port with regular updates from C# to TS.

Good luck with the project!

**Comment by jasongdove** [comment]
Forgot one of the important non-normalization changes I ported over: proxy or redirect to hide the plex tokens from log files (ffmpeg arguments). Much easier for users to feel safe about sharing logs that way; DTV logs can be littered with plex direct URLs including tokens.

https://github.com/jasongdove/dizquetv/pull/9

**Comment by chrisbenincasa** [comment]
Wow, nice work! I definitely need to investigate Ersatz's ffmpeg stuff a bit more. The organization there is really nice as well; the old dizque code could _definitely_ use a revamp...it would go a long way in adding improvements to it without breaking stuff, too.

I opened up #183 to track the Plex token issues.
